ref/general/authorization.adoc

```diff
@@ -8,42 +8,79 @@
+[source,java]
+---
+<application-bnd>
+   <security-role name="admin">
+     <group name="Manager" />
+     <group name="TeamLead" />
+   </security-role>
+   <security-role name="user">
+     <group name="Employee" />
+   </security-role>
+</application-bnd>
+ ---
 
-In cases where an SSO provider is configured, the user’s credentials are retrieved from a token (such as a JSON Web Token). This token can then propagate the identity of the authenticated user to application resources in an efficient, self-contained way
 
-In some use cases, an application needs to know only whether a user is authenticated or not. In such cases, configuration can be simplified by specifying `"**"` as the role name in the `@RolesAllowed` annotation or XML configuration. This value denotes that any authenticated user can access a resource. It is also possible to authorize all users by setting the `@PermitAll` annotation, which specifies that a resource is accessible to everyone, even users who are not authenticated.
+The `name` attribute of the `<security-role>` element corresponds to a security role specified in the application and the group names correspond to the user groups to which authenticated users belong.
+If the role names match the group names exactly, the server maps the group to the role with a matching name by default.
+If you use this default mapping, you do not need specify an `<application-bnd>` element in the `server.xml` file.
 
-As an alternative to the `@RolesAllowed` annotation, you can define the `@ServletSecurity` annotation in the `HomeServlet.java` file with the `@HttpConstraint` annotation and set the `rolesAllowed` attribute to the permissible roles. In this configuration, role mappings are defined either in `application-bnd` stanzas in the `server.xml` file or in an `ibm-application-bnd.xml` file in the application.
+In some cases, an application needs to know only whether a user is authenticated or not.
+In such cases, you can specify `**` as the role name in the `@RolesAllowed` annotation.
+This value denotes that any authenticated user can access a resource.
+Alternatively, you can authorize all users by setting the `@PermitAll` annotation, which specifies that anyone can access a resource, even users who are not authenticated.
 
-== Authorize requests between web services
+The `@RolesAllowed` annotation typically specifies role-based access to business logic application resources, such as calls to a database.
+To control authorization for front-end resources, such as HTTP calls, you can configure security roles at the servlet level.
+You can specify security roles in the servlet code by configuring the `@ServletSecurity` annotation. Set the `rolesAllowed` attribute within the `@HttpConstraint` or `@HttpMethodConstraint` annotation to the permissible roles, as shown in the following example:
```

Comment:
Should this be "within the `@HttpConstraint` or `@HttpMethodConstraint` annotation**s**"?